src/language/dafnyLanguageClient.ts

```diff
@@ -61,27 +66,39 @@ export class DafnyLanguageClient extends LanguageClient {
       documentSelector: [ DafnyDocumentFilter ],
       diagnosticCollectionName: LanguageServerId
     };
-    return new DafnyLanguageClient(LanguageServerId, LanguageServerName, serverOptions, clientOptions);
+    return new DafnyLanguageClient(LanguageServerId, LanguageServerName, serverOptions, clientOptions, outputChannel);
   }
 
   public onGhostDiagnostics(callback: (params: IGhostDiagnosticsParams) => void): Disposable {
-    return this.onNotification('dafny/ghost/diagnostics', callback);
+    return this.onLoggedNotification('dafny/ghost/diagnostics', callback);
   }
 
   public onCompilationStatus(callback: (params: ICompilationStatusParams) => void): Disposable {
-    return this.onNotification('dafny/compilation/status', callback);
+    return this.onLoggedNotification('dafny/compilation/status', callback);
   }
 
   public onServerVersion(callback: (version: string) => void): Disposable {
-    return this.onNotification('dafnyLanguageServerVersionReceived', callback);
+    return this.onLoggedNotification('dafnyLanguageServerVersionReceived', callback);
   }
 
   // TODO Legacy verification status messages
   public onVerificationStarted(callback: (params: IVerificationStartedParams) => void): Disposable {
-    return this.onNotification('dafny/verification/started', callback);
+    return this.onLoggedNotification('dafny/verification/started', callback);
   }
 
   public onVerificationCompleted(callback: (params: IVerificationCompletedParams) => void): Disposable {
-    return this.onNotification('dafny/verification/completed', callback);
+    return this.onLoggedNotification('dafny/verification/completed', callback);
+  }
+
+  public onLoggedNotification<Params, Return>(route: string, callback: (params: Params) => Return): Disposable {
```

Comment:
Prefix type arguments with `T`. And use `TResult` instead of `TReturn`.
Preferably `private`


You may add the following rule to `.eslintrc.json`:
```json
    "@typescript-eslint/naming-convention": [
      "warn",
      {
        "selector": "typeParameter",
        "format": [ "PascalCase" ],
        "prefix": [ "T" ]
      }
    ]
```

Reply:
I added this rule and restarted VSCode but it did not seem to work